### PR TITLE
feat(calendar):Add functionality to allow users to select a specific year and month.

### PR DESCRIPTION
**Pull request by auzeonfung**
## Support for selecting the year and month using the select component.

- The year and month can now be specified directly using the dropdown selector!
- Filled in the missing styles.

image

image

` dropdown-buttons `

image

` dropdown `

image

` buttons `

Related to:

#320 #546 #509 #880 

### Usage:
```
<Calendar
    initialFocus
    mode="range" // range | single
    captionLayout="dropdown-buttons" 
    // Optionally, enables the year and month drop-down selector.
    // also: dropdown-buttons | dropdown | buttons
    fromYear={2022} 
    toYear={2023} 
    defaultMonth={date?.from}
    selected={date}
    onSelect={setDate}
    numberOfMonths={2}
    {...props}
/>
```

### CODE REVIEW

**Comment by shadcn** [comment]
Wow. Thank you. Let's review.

File: apps/www/registry/default/ui/calendar.tsx [hunk 1]
```diff
@@ -3,9 +3,9 @@
 import * as React from "react"
 import { ChevronLeft, ChevronRight } from "lucide-react"
```
Comment (Gary50613) [inline apps/www/registry/default/ui/calendar.tsx:4]: should be `ChevronLeftIcon` and `ChevronRightIcon` here?

**Comment by tanishbaansal** [comment]
Hey @shadcn can you merge this one , it would be really useful

File: apps/www/registry/default/ui/calendar.tsx (hunk 1 above)
Comment (tanishbaansal) [reply apps/www/registry/default/ui/calendar.tsx:4]: Its correct - 
image

https://lucide.dev/icons/chevron-left

**Comment by DarkAbhi** [comment]
@shadcn Any updates on this one?

**Review by Connor-lipalater — APPROVED** [review]

**Pull request closed without merging**